core/blockchain_impl.go

```diff
@@ -447,7 +447,7 @@ func VerifyIncomingReceipts(blockchain BlockChain, block *types.Block) error {
 			}
 		}
 
-		if err := blockchain.Validator().ValidateCXReceiptsProof(cxp); err != nil {
+		if err := NewBlockValidator(blockchain).ValidateCXReceiptsProof(cxp); err != nil {
```

Comment:
Would it be better to keep `bc.Validator()` without mutex as a function for cases like this?

Reply:
@MaxMustermann2 just curious, could you explain your rationale of doing so ?